tests/ostree-spawn.c

```diff
@@ -112,6 +114,36 @@ ostree_init (GFile *repo,
                                     error);
 }
 
+static void
+copy_additional_metadata_args_from_hashtable (GArray *cmd_args,
+                                              GHashTable *metadata,
+                                              GPtrArray *cmd_args_membuf)
+{
+  gpointer key;
+  gpointer value;
+  GHashTableIter iter;
+
+  if (metadata == NULL)
+    return;
+
+  g_hash_table_iter_init (&iter, metadata);
+
+  while (g_hash_table_iter_next (&iter, &key, &value))
+    {
+      gchar *formatted_metadata_string = g_strdup_printf ("%s=%s",
```

Comment:
`g_autofree`

Reply:
Perhaps `g_autofree` + `g_steal_pointer()` somewhere appropriate then? I haven’t looked at the code to see what would fit, but I would much prefer to make all ownership transfers explicit using those two annotations, rather than have them implicit in the code (and thus easier to break when refactoring).